p1_runner/output_server.py

```diff
@@ -203,9 +203,13 @@ def _run_tcp(self):
                         'Unexpected error from TCP socket:\r%s' % traceback.format_exc())
                 break
 
-            data = client.recv(1024)
-            if data and self.incoming_data_callback is not None:
-                self.incoming_data_callback(data)
+            try:
+                data = client.recv(1024)
+                if data and self.incoming_data_callback is not None:
+                    self.incoming_data_callback(data)
+            except (BrokenPipeError, ConnectionResetError):
+                self.logger.error('Reconnecting tcp://'+str(addr[0])+':'+str(addr[1]))
```

Comment:
I don't think this is right though? This app is the server. If the client disconnects, we don't know if they're reconnecting or not. (It also should be a complete sentence ideally.) This should just say "Output client tcp://... disconnected.". It should probably just be an `info()` since they're totally allowed to disconnect whenever they want, that's not an error.

Minor but also need spaces around the + operators. Or better would be to use either `%` or an `f''` string instead.